### ISSUE TITLE
`pygls` crashes in python 3.12b1

### ISSUE THREAD

**Issue by karthiknadig** [openlawlibrary/pygls#341]
```
2023-06-16 08:24:41.930 [info]     from pygls import server, uris, workspace
  File "/<redacted>/.vscode/extensions/ms-python.black-formatter-2023.2.0/bundled/libs/pygls/server.py", line 27, in <module>
    from pygls.lsp import ConfigCallbackType, ShowDocumentCallbackType
  File "/<redacted>/.vscode/extensions/ms-python.black-formatter-2023.2.0/bundled/libs/pygls/lsp/__init__.py", line 36, in <module>
    from typeguard import check_type
  File "/<redacted>/.vscode/extensions/ms-python.black-formatter-2023.2.0/bundled/libs/typeguard/__init__.py", line 27, in <module>
    from typing_extensions import Literal
  File "/<redacted>/.vscode/extensions/ms-python.black-formatter-2023.2.0/bundled/libs/typing_extensions.py", line 1174, in <module>

2023-06-16 08:24:41.930 [info]     class TypeVar(typing.TypeVar, _DefaultMixin, _root=True):
TypeError: type 'typing.TypeVar' is not an acceptable base type

```

**Comment by alcarney** [comment]
Looking at the traceback, isn't the issue in `typing-extensions`?

**Comment by karthiknadig** [comment]
@alcarney I think it is `typing-extensions`, I created this for tracking.

**Comment by karthiknadig** [comment]
Looks like this is addressed in `typing_extensions` 4.6.0. we can close this.
